### PR TITLE
Add optional "planb_ignore" key to allow skipping manifest packages.

### PR DESCRIPTION
**Pull request by nguyen-phillip**
Also manifest validation now correctly treats "sha256" key as optional.

### CODE REVIEW

File: planb/PBManifest.m [hunk 1]
```diff
@@ -135,6 +137,11 @@ - (BOOL)validatePackage:(id)object {
       return NO;
     }
   }
+  if (package[kPlanBIgnoreKey] && ![package[kPlanBIgnoreKey] isKindOfClass:[NSNumber class]]) {
```
Comment (tburgin) [inline planb/PBManifest.m:141]: The "package[kPlanBIgnoreKey]" check is redundant. Sending a isKindOfClass: message to nil is safe and will return nil (NO).

File: planb/PBManifest.m [hunk 2]
```diff
@@ -159,11 +166,15 @@ - (BOOL)validateTrack:(id)object {
...
-    if (![item[key] isKindOfClass:[NSString class]]) {
-      PBLog(@"Invalid manifest -- track value for %@ key is not a string: %@", key, item[key]);
-      return NO;
-    }
+  if (![item[kFilenameKey] isKindOfClass:[NSString class]]) {
+    PBLog(@"Invalid manifest -- track value for %@ key is not a string: %@", kFilenameKey,
+          item[kFilenameKey]);
+    return NO;
+  }
+  if (item[kSHA256Key] && ![item[kSHA256Key] isKindOfClass:[NSString class]]) {
```
Comment (tburgin) [inline planb/PBManifest.m:176]: same

File: planb/PBManifest.m (hunk 1 above)
Comment (nguyen-phillip) [reply planb/PBManifest.m:141]: The first check is to see if the key exists -- it's optional so I don't want the second check to fail if there is no planb_ignore key in the manifest.

File: planb/PBManifest.m (hunk 2 above)
Comment (nguyen-phillip) [reply planb/PBManifest.m:176]: The sha256 key is also optional.  I thought about maybe making a more generic validator object for all of these that would include the key name, it's desired class, and whether or not it was optional, then looping over everything, but it seemed like a lot of overhead to set up.  I could maybe make it into a function call like ValidateKey(dict, name, class, optional).  But then the error logs would lose some of their specificity.  So I don't know.

File: planb/PBManifest.m (hunk 1 above)
Comment (tburgin) [reply planb/PBManifest.m:141]: Ah, gotcha

File: planb/PBManifest.m (hunk 2 above)
Comment (tburgin) [reply planb/PBManifest.m:176]: This is fine, maybe just add a comment. The extra check is for logic not safety.

Comment (nguyen-phillip) [reply planb/PBManifest.m:176]: Done.

**Comment by nguyen-phillip** [comment]
PTAL

**Review by tburgin — APPROVED** [review]

**Pull request merged**